rcl/src/rcl/node.c

```diff
@@ -179,7 +211,47 @@ rcl_node_init(
   }
   // actual domain id
   node->impl->actual_domain_id = domain_id;
-  node->impl->rmw_node_handle = rmw_create_node(name, local_namespace_, domain_id);
+
+  const char * ros_enable_security = NULL;
+  const char * ros_enforce_security = NULL;
+
+  if (rcutils_get_env(sros_enable_security_var_name, &ros_enable_security)) {
```

Comment:
Since `rcutils_get_env` is not thread safe it make all code using that function not thread safe which I think is a problem here. I think we need a thread safe version of this if we want to use it throughout the code like here.